artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPLargeMessageReader.java

```diff
@@ -101,7 +104,7 @@ public Message readBytes(Delivery delivery) throws Exception {
       final AMQPLargeMessage result;
 
       if (!delivery.isPartial()) {
-         currentMessage.releaseResources(true, true);
+         currentMessage.releaseResources(largeMessageSync, true);
```

Comment:
You don't need to track this value, just call the accessor in the Session conetxt